crates/dynamodb-streams/src/stream.rs

```diff
@@ -0,0 +1,169 @@
+use crate::client_sdk::SDKClient;
+use crate::stream_state::{RecordBatch, StreamState};
+use crate::{Error, Result, StreamResult};
+use aws_sdk_dynamodbstreams::types::ShardIteratorType;
+use futures::{Stream, future::join_all};
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+use tokio::{
+    sync::mpsc,
+    time::{Duration, sleep},
+};
+use util::retry_strategy::{Backoff, RetryBackoff};
+
+#[derive(Debug)]
+pub struct DynamodbStreamProducer {
+    pub stream_arn: String,
+    pub state: StreamState,
+    pub interval: Option<Duration>,
+    pub sender: mpsc::Sender<StreamResult>,
+    pub client: Arc<SDKClient>,
+    pub retry_strategy: RetryBackoff,
+}
+
+impl DynamodbStreamProducer {
+    async fn iterate(&mut self) -> Result<Vec<RecordBatch>> {
```

Comment:
maybe call this `collect` to align with what DataFusion calls their stream collector?